spec.bs

```diff
@@ -512,6 +517,11 @@ The <dfn for=Navigator method>runAdAuction(|config|)</dfn> method steps are:
 1. Let |auctionConfig| be the result of running [=validate and convert auction ad config=] with
   |config| and true.
 1. If |auctionConfig| is failure, then [=exception/throw=] a {{TypeError}}.

```

Comment:
(lt doesn't look like `auctionConfig` is ever set to failure, reading thru https://wicg.github.io/turtledove/#validate-and-convert-auction-ad-config, but instead has a bunch of conditions that throw TypeErrors - am I reading this right, @JensenPaul? (happy to file a non-blocking bug if so))

Reply:
That seems like the case.
Another non-blocking bug is likely where it is invoked for component auctions. It doesn't seem to handle failures:
https://wicg.github.io/turtledove/#ref-for-validate-and-convert-auction-ad-config%E2%91%A0:~:text=Let%20componentAuction%20be,component%20auctions.

And whether https://wicg.github.io/turtledove/#validate-and-convert-auction-ad-config throwing exceptions is acceptable for component auctions - if one fails, the others can still proceed, right?

Reply:
It seems to be a mistake in the spec that needs to be fixed. In [our code](https://crsrc.org/c/third_party/blink/renderer/modules/ad_auction/navigator_auction.cc;drc=eef4762779d05708d5dfc7d5fe4ea16288069a35;l=2150), a bad component auction config will cause the whole auction config being an empty one. And the runAdAuction [returns](https://crsrc.org/c/third_party/blink/renderer/modules/ad_auction/navigator_auction.cc;drc=eef4762779d05708d5dfc7d5fe4ea16288069a35;l=3155) without running the auction.

Reply:
I've filed https://github.com/WICG/turtledove/issues/788 related to this.